routes/conversation.js

```diff
@@ -1,16 +1,78 @@
-const express = require("express");
+const express = require('express');
 const router = express.Router();
+const auth = require('../middleware/auth');
 const Conversation = require('../models/Conversation');
 
-router.post('/add', (req, res) => {
-    const conversation = new Conversation({
-        members: req.body.members, // array
-        messages: []
-    })
-})
+//POST a new conversation
+router.post('/add', auth, async (req, res) => {
+  const members = req.body;
 
-router.delete('/delete/:id', (req, res) => {
-        
-})
+  const memberIDs = members.map((member) => member._id);
 
-module.exports = router;
\ No newline at end of file
+  try {
+    //check to see if conversation exists
+    let conversation = await Conversation.findOne({
+      'members._id': { $all: memberIDs },
+    });
+
+    //if conversation already exists
+    if (conversation) return res.status(409).json(conversation);
+
+    conversation = new Conversation({
+      members,
+      messages: [],
+    });
+
+    await conversation.save();
+    return res.status(200).json(conversation);
+  } catch (err) {
+    console.error(err);
+    res.status(500).send('Server error');
+  }
+});
+
+//GET a single conversation
+router.get('/room/:memberIDs', auth, async (req, res) => {
+  const memberIDsRaw = req.params.memberIDs;
```

Comment:
i believe you can access the query directly like

```suggestion
  const memberIDsRaw = req.params.members;
```

and this should automatically give you an array of members you're searching for

Reply:
resource: https://stackabuse.com/get-query-strings-and-parameters-in-express-js/